Do we need more changes? I'm seeing different missing method exceptions now in the Helix logs

```
C:\h\w\C36F0AA6\p\d\sdk\6.0.400-ci\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.targets(628,5): error MSB4018: System.MissingMethodException: Method not found: 'System.Collections.Immutable.ImmutableArray`1<System.Reflection.Metadata.CustomAttributeTypedArgument`1<!0>> System.Reflection.Metadata.CustomAttributeValue`1.get_FixedArguments()
```

--- a/eng/Versions.props
+++ b/eng/Versions.props
@@ -98,6 +98,11 @@
   <PropertyGroup>
     <!-- Dependencies from https://github.com/Microsoft/msbuild -->
     <MicrosoftBuildPackageVersion>17.3.0-preview-22306-01</MicrosoftBuildPackageVersion>
+    <!-- .NET Framework-targeted tasks will need to run in an MSBuild that is older than the very latest,
+          so target one that matches the version in minimumMSBuildVersion.
+
+          This avoids the need to juggle references to packages that have been updated in newer MSBuild. -->
+    <MicrosoftBuildPackageVersion Condition=" '$([MSBuild]::GetTargetFrameworkIdentifier($(TargetFramework)))' == '.NETFramework' ">$([System.IO.File]::ReadAllText('$(RepoRoot)\src\Layout\redist\minimumMSBuildVersion').Trim())</MicrosoftBuildPackageVersion>
     <MicrosoftBuildFrameworkPackageVersion>$(MicrosoftBuildPackageVersion)</MicrosoftBuildFrameworkPackageVersion>
     <MicrosoftBuildRuntimePackageVersion>$(MicrosoftBuildPackageVersion)</MicrosoftBuildRuntimePackageVersion>
     <MicrosoftBuildLocalizationPackageVersion>17.3.0-preview-22306-01</MicrosoftBuildLocalizationPackageVersion>

--- a/src/Tasks/Microsoft.NET.Build.Tasks/Microsoft.NET.Build.Tasks.csproj
+++ b/src/Tasks/Microsoft.NET.Build.Tasks/Microsoft.NET.Build.Tasks.csproj
@@ -65,17 +65,9 @@
         MSBuild depends on a new version of SRM/SCI; HostModel a very old one.
         In order to keep working with MSBuild 17.2, which had binding redirects
         for SCI 0-5.0.0.0 but not 6, this task must compile against 5, even
-        though that's not transitively correct. Nothing uses updated API
-        surface so it all works ok.
-
-        To accomplish this, explicitly reference the current version of the
-        packages, but exclude assets so it doesn't get passed to the compiler.
-        Also download an old version and (later in the target ReferenceOlderSCIandSRM)
-        pass them to the compiler. -->
-    <PackageReference Include="System.Reflection.Metadata" Version="$(SystemReflectionMetadataVersion)" IncludeAssets="none" />
-    <PackageReference Include="System.Collections.Immutable" Version="6.0.0" IncludeAssets="none" />
-    <PackageDownload Include="System.Collections.Immutable" Version="[5.0.0]" GeneratePathProperty="true" />
-    <PackageDownload Include="System.Reflection.Metadata" Version="[5.0.0]" GeneratePathProperty="true" />
+        though that's not the very latest. -->
+    <PackageReference Include="System.Collections.Immutable" Version="5.0.0" />
+    <PackageReference Include="System.Reflection.Metadata" Version="5.0.0" />
   </ItemGroup>
 
   <!-- These are loaded from the CLI's copy on .NET Core, we don't need to duplicate them on disk -->
@@ -113,15 +105,6 @@
     <None Include="..\Common\Resources\xlf\**\*" LinkBase="Resources\xlf" />
     <UpToDateCheckInput Include="@(None)" />
   </ItemGroup>
-
-  <Target Name="ReferenceOlderSCIandSRM" AfterTargets="ResolveAssemblyReferences" Condition="'$(TargetFrameworkIdentifier)' == '.NETFramework'">
-    <ItemGroup>
-      <ReferencePath Include="$(NuGetPackageRoot)system.collections.immutable\5.0.0\lib\net461\System.Collections.Immutable.dll" />
-      <ReferencePath Include="$(NuGetPackageRoot)system.reflection.metadata\5.0.0\lib\net461\System.Reflection.Metadata.dll" />
-      <ReferenceCopyLocalPaths Include="$(NuGetPackageRoot)system.collections.immutable\5.0.0\lib\net461\System.Collections.Immutable.dll" />
-      <ReferenceCopyLocalPaths Include="$(NuGetPackageRoot)system.reflection.metadata\5.0.0\lib\net461\System.Reflection.Metadata.dll" />
-    </ItemGroup>
-  </Target>
 
   <Target Name="PrepareAdditionalFilesToLayout" BeforeTargets="AssignTargetPaths">
     <PropertyGroup>

